# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/cs_clip_box_shadow.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/cs_clip_box_shadow.glsl
--- a/gfx/wr/webrender/res/cs_clip_box_shadow.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/cs_clip_box_shadow.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -4,13 +4,13 @@
 
 #include shared,clip_shared
 
-varying vec4 vLocalPos;
-varying vec2 vUv;
-flat varying vec4 vUvBounds;
-flat varying vec4 vEdge;
-flat varying vec4 vUvBounds_NoClamp;
+varying highp vec4 vLocalPos;
+varying highp vec2 vUv;
+flat varying highp vec4 vUvBounds;
+flat varying mediump vec4 vEdge;
+flat varying highp vec4 vUvBounds_NoClamp;
 // Clip mode. Packed in to a vector to avoid bug 1630356.
-flat varying vec2 vClipMode;
+flat varying mediump vec2 vClipMode;
 
 #define MODE_STRETCH        0
 #define MODE_SIMPLE         1